modules/drivers/mongo/src/metabase/driver/mongo/util.clj

```diff
@@ -89,6 +88,36 @@
     :else                         (throw (Exception. (str "with-mongo-connection failed: bad connection details:"
                                                           (:details database))))))
 
+(defn- connect-via-uri-w-opts
+  [^MongoClientURI uri ^MongoClient conn]
+  (if-let [dbName (.getDatabase uri)]
+    (.getDB conn dbName)
+    (throw (IllegalArgumentException. "No database name specified in URI. Monger requires a database to be explicitly configured."))))
+
+(defn- connect [host port user authdb pass dbname ssl additional-options]
+  (let [server-address (mg/server-address host port)
+        credentials (when user
+                      (mcred/create user authdb pass))
+        ^MongoClientOptions$Builder opts (connection-options-builder :ssl? ssl, :additional-options additional-options)
+        do-connect (partial mg/connect server-address (-> opts .build))
+        mongo-client (if credentials
+                       (do-connect credentials)
+                       (do-connect))
+        db (mg/get-db mongo-client dbname)]
+    [mongo-client db]))
+
+(defn- connect-srv [host port user authdb pass dbname ssl additional-options]
```

Comment:
IMO having big 8-arg functions like this is hard to work with,

why not just make it a map instead and make your argslists be

```clj
[{:keys [host port user authdb pass dbname ssl additional-options]}]
```

(this applies to both `connect` and `connect-srv`

Reply:
In general we try to follow the Clojure Style Guide as much as possible, e.g.

https://github.com/bbatsov/clojure-style-guide#function-positional-parameter-limit

for not using too many positional params

Reply:
Fixed